### PR TITLE
Add tidb_parser{,_test}.go

### PR DESCRIPTION
**Pull request by wangkuiyi**

### CODE REVIEW

File: parser/tidb/tidb_parser.go [hunk 1]
```diff
@@ -0,0 +1,30 @@
+package tidb
+
+import (
+	"fmt"
+	"regexp"
+	"strings"
+
+	"github.com/pingcap/parser"
+	_ "github.com/pingcap/tidb/types/parser_driver"
```
Comment (tonyyang-svail) [inline parser/tidb/tidb_parser.go]: Do we need to import the driver?

**Comment by tonyyang-svail** [comment]
CI failed on golint:
```
go lint..................................................................Failed
hookid: go-lint
parser/tidb/tidb_parser.go:9:2: a blank import should be only in a main or test package, or have a comment justifying it
parser/tidb/tidb_parser.go:16:1: error should be the last type when returning multiple items
Found 2 lint suggestions; failing.
```

File: parser/tidb/tidb_parser_test.go [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestTiDBParser(t *testing.T) {
+	a := assert.New(t)
+
+	e, i := Parse("select * from tbl where id = 1")
```
Comment (typhoonzero) [inline parser/tidb/tidb_parser_test.go]: Maybe add test case for wrong standard SQL statement and wrong SQLFlow statement to test if the parser can catch expected error in these cases?

File: parser/tidb/tidb_parser.go (hunk 1 above)
Comment (wangkuiyi) [reply parser/tidb/tidb_parser.go]: Followed https://github.com/pingcap/parser/blob/master/parser_example_test.go

**Comment by wangkuiyi** [comment]
This time, the CI failure seems irrelevant to this PR:

```
--- FAIL: TestWriteAndRead (5.82s)
    sqlfile_test.go:74: 
        	Error Trace:	sqlfile_test.go:74
        	Error:      	Received unexpected error:
        	            	writer flush failed: flush to sqlfs_test.unitest6129484611666145821, error:Error from server: TStatus({StatusCode:ERROR_STATUS InfoMessages:[*org.apache.hive.service.cli.HiveSQLException:Error while processing statement: FAILED: Execution Error, return code 2 from org.apache.hadoop.hive.ql.exec.mr.MapRedTask:17:16 org.apache.hive.service.cli.operation.Operation:toSQLException:Operation.java:380 org.apache.hive.service.cli.operation.SQLOperation:runQuery:SQLOperation.java:257 org.apache.hive.service.cli.operation.SQLOperation:runInternal:SQLOperation.java:293 org.apache.hive.service.cli.operation.Operation:run:Operation.java:320 org.apache.hive.service.cli.session.HiveSessionImpl:executeStatementInternal:HiveSessionImpl.java:530 org.apache.hive.service.cli.session.HiveSessionImpl:executeStatement:HiveSessionImpl.java:506 org.apache.hive.service.cli.CLIService:executeStatement:CLIService.java:280 org.apache.hive.service.cli.thrift.ThriftCLIService:ExecuteStatement:ThriftCLIService.java:531 org.apache.hive.service.rpc.thrift.TCLIService$Processor$ExecuteStatement:getResult:TCLIService.java:1437 org.apache.hive.service.rpc.thrift.TCLIService$Processor$ExecuteStatement:getResult:TCLIService.java:1422 org.apache.thrift.ProcessFunction:process:ProcessFunction.java:39 org.apache.thrift.TBaseProcessor:process:TBaseProcessor.java:39 org.apache.hive.service.auth.TSetIpAddressProcessor:process:TSetIpAddressProcessor.java:56 org.apache.thrift.server.TThreadPoolServer$WorkerProcess:run:TThreadPoolServer.java:286 java.util.concurrent.ThreadPoolExecutor:runWorker:ThreadPoolExecutor.java:1149 java.util.concurrent.ThreadPoolExecutor$Worker:run:ThreadPoolExecutor.java:624 java.lang.Thread:run:Thread.java:748] SqlState:0xc00007d300 ErrorCode:0xc000088e48 ErrorMessage:0xc00007d310})
        	Test:       	TestWriteAndRead
    sqlfile_test.go:75: 
        	Error Trace:	sqlfile_test.go:75
        	Error:      	Not equal: 
        	            	expected: 4097
        	            	actual  : 4093
        	Test:       	TestWriteAndRead
    sqlfile_test.go:94: 
        	Error Trace:	sqlfile_test.go:94
        	Error:      	Not equal: 
        	            	expected: 4098
        	            	actual  : 4094
        	Test:       	TestWriteAndRead
    sqlfile_test.go:96: 
        	Error Trace:	sqlfile_test.go:96
        	Error:      	Not equal: 
        	            	expected: 4097
        	            	actual  : 4093
        	Test:       	TestWriteAndRead
FAIL
FAIL	github.com/sql-machine-learning/sqlflow/sqlfs	15.058s
```

Anyone knows the reason? @typhoonzero @tonyyang-svail @weiguoz

**Comment by weiguoz** [comment]
> This time, the CI failure seems irrelevant to this PR:
> 
> ```
> --- FAIL: TestWriteAndRead (5.82s)
>     sqlfile_test.go:74: 
>         	Error Trace:	sqlfile_test.go:74
>         	Error:      	Received unexpected error:
>         	            	writer flush failed: flush to sqlfs_test.unitest6129484611666145821, error:Error from server: TStatus({StatusCode:ERROR_STATUS InfoMessages:[*org.apache.hive.service.cli.HiveSQLException:Error while processing statement: FAILED: Execution Error, return code 2 from org.apache.hadoop.hive.ql.exec.mr.MapRedTask:17:16 org.apache.hive.service.cli.operation.Operation:toSQLException:Operation.java:380 org.apache.hive.service.cli.operation.SQLOperation:runQuery:SQLOperation.java:257 org.apache.hive.service.cli.operation.SQLOperation:runInternal:SQLOperation.java:293 org.apache.hive.service.cli.operation.Operation:run:Operation.java:320 org.apache.hive.service.cli.session.HiveSessionImpl:executeStatementInternal:HiveSessionImpl.java:530 org.apache.hive.service.cli.session.HiveSessionImpl:executeStatement:HiveSessionImpl.java:506 org.apache.hive.service.cli.CLIService:executeStatement:CLIService.java:280 org.apache.hive.service.cli.thrift.ThriftCLIService:ExecuteStatement:ThriftCLIService.java:531 org.apache.hive.service.rpc.thrift.TCLIService$Processor$ExecuteStatement:getResult:TCLIService.java:1437 org.apache.hive.service.rpc.thrift.TCLIService$Processor$ExecuteStatement:getResult:TCLIService.java:1422 org.apache.thrift.ProcessFunction:process:ProcessFunction.java:39 org.apache.thrift.TBaseProcessor:process:TBaseProcessor.java:39 org.apache.hive.service.auth.TSetIpAddressProcessor:process:TSetIpAddressProcessor.java:56 org.apache.thrift.server.TThreadPoolServer$WorkerProcess:run:TThreadPoolServer.java:286 java.util.concurrent.ThreadPoolExecutor:runWorker:ThreadPoolExecutor.java:1149 java.util.concurrent.ThreadPoolExecutor$Worker:run:ThreadPoolExecutor.java:624 java.lang.Thread:run:Thread.java:748] SqlState:0xc00007d300 ErrorCode:0xc000088e48 ErrorMessage:0xc00007d310})
>         	Test:       	TestWriteAndRead
>     sqlfile_test.go:75: 
>         	Error Trace:	sqlfile_test.go:75
>         	Error:      	Not equal: 
>         	            	expected: 4097
>         	            	actual  : 4093
>         	Test:       	TestWriteAndRead
>     sqlfile_test.go:94: 
>         	Error Trace:	sqlfile_test.go:94
>         	Error:      	Not equal: 
>         	            	expected: 4098
>         	            	actual  : 4094
>         	Test:       	TestWriteAndRead
>     sqlfile_test.go:96: 
>         	Error Trace:	sqlfile_test.go:96
>         	Error:      	Not equal: 
>         	            	expected: 4097
>         	            	actual  : 4093
>         	Test:       	TestWriteAndRead
> FAIL
> FAIL	github.com/sql-machine-learning/sqlflow/sqlfs	15.058s
> ```
> 
> Anyone knows the reason? @typhoonzero @tonyyang-svail @weiguoz

This caused by `gohive`, I filed an issue here([Select with an order by reports error](https://github.com/sql-machine-learning/gohive/issues/39)).
But I would like to do sorting in `sqlflow` code for a quick fix

**Comment by tonyyang-svail** [comment]
> Anyone knows the reason? @typhoonzero @tonyyang-svail @weiguoz

This is due to the flaky tests on Hive. @weiguoz is looking into it.

**Comment by weiguoz** [comment]
> This time, the CI failure seems irrelevant to this PR:
> 
> ```
> --- FAIL: TestWriteAndRead (5.82s)
>     sqlfile_test.go:74: 
>         	Error Trace:	sqlfile_test.go:74
>         	Error:      	Received unexpected error:
>         	            	writer flush failed: flush to sqlfs_test.unitest6129484611666145821, error:Error from server: TStatus({StatusCode:ERROR_STATUS InfoMessages:[*org.apache.hive.service.cli.HiveSQLException:Error while processing statement: FAILED: Execution Error, return code 2 from org.apache.hadoop.hive.ql.exec.mr.MapRedTask:17:16 org.apache.hive.service.cli.operation.Operation:toSQLException:Operation.java:380 org.apache.hive.service.cli.operation.SQLOperation:runQuery:SQLOperation.java:257 org.apache.hive.service.cli.operation.SQLOperation:runInternal:SQLOperation.java:293 org.apache.hive.service.cli.operation.Operation:run:Operation.java:320 org.apache.hive.service.cli.session.HiveSessionImpl:executeStatementInternal:HiveSessionImpl.java:530 org.apache.hive.service.cli.session.HiveSessionImpl:executeStatement:HiveSessionImpl.java:506 org.apache.hive.service.cli.CLIService:executeStatement:CLIService.java:280 org.apache.hive.service.cli.thrift.ThriftCLIService:ExecuteStatement:ThriftCLIService.java:531 org.apache.hive.service.rpc.thrift.TCLIService$Processor$ExecuteStatement:getResult:TCLIService.java:1437 org.apache.hive.service.rpc.thrift.TCLIService$Processor$ExecuteStatement:getResult:TCLIService.java:1422 org.apache.thrift.ProcessFunction:process:ProcessFunction.java:39 org.apache.thrift.TBaseProcessor:process:TBaseProcessor.java:39 org.apache.hive.service.auth.TSetIpAddressProcessor:process:TSetIpAddressProcessor.java:56 org.apache.thrift.server.TThreadPoolServer$WorkerProcess:run:TThreadPoolServer.java:286 java.util.concurrent.ThreadPoolExecutor:runWorker:ThreadPoolExecutor.java:1149 java.util.concurrent.ThreadPoolExecutor$Worker:run:ThreadPoolExecutor.java:624 java.lang.Thread:run:Thread.java:748] SqlState:0xc00007d300 ErrorCode:0xc000088e48 ErrorMessage:0xc00007d310})
>         	Test:       	TestWriteAndRead
>     sqlfile_test.go:75: 
>         	Error Trace:	sqlfile_test.go:75
>         	Error:      	Not equal: 
>         	            	expected: 4097
>         	            	actual  : 4093
>         	Test:       	TestWriteAndRead
>     sqlfile_test.go:94: 
>         	Error Trace:	sqlfile_test.go:94
>         	Error:      	Not equal: 
>         	            	expected: 4098
>         	            	actual  : 4094
>         	Test:       	TestWriteAndRead
>     sqlfile_test.go:96: 
>         	Error Trace:	sqlfile_test.go:96
>         	Error:      	Not equal: 
>         	            	expected: 4097
>         	            	actual  : 4093
>         	Test:       	TestWriteAndRead
> FAIL
> FAIL	github.com/sql-machine-learning/sqlflow/sqlfs	15.058s
> ```
> 
> Anyone knows the reason? @typhoonzero @tonyyang-svail @weiguoz

Access to `hive` is not stable yet. After several tests in locally, I found that the error was random.
For similar errors, I filed an [issue here](https://github.com/sql-machine-learning/gohive/issues/39).

Meanwhile, I restarted the CI and got a pass for this PR.

**Review by tonyyang-svail — APPROVED** [review]
LGTM.

**Review by typhoonzero — COMMENTED** [review]
LGTM++

**Review by weiguoz — APPROVED** [review]

**Pull request merged**